contracts/upgradeable_contracts/omnibridge_nft/modules/forwarding_rules/NFTForwardingRulesManager.sol

```diff
@@ -15,8 +15,15 @@ contract NFTForwardingRulesManager is OmnibridgeModule {
 
     event ForwardingRuleUpdated(address token, address sender, address receiver, int256 lane);
 
-    // solhint-disable-next-line no-empty-blocks
-    constructor(IOwnable _mediator) OmnibridgeModule(_mediator) {}
+    /**
+     * @dev Initializes this module contract. Intended to be called only once through the proxy pattern.
+     * @param _mediator address of the Omnibridge contract working with this module.
+     */
+    function initialize(IOwnable _mediator) external {
```

Comment:
We have quite complex structure of the nested contracts. Do you have any recommendation how we can preserve the order of storage slots during next modifications of the contracts. The eternal storage guaranteed that occupying of a new storage slot does not affect existing slots. With the new approach the developers are responsible to track the order.

Reply:
I think that internal structure of modules is much simpler than structure of mediator. That's why it is not necessary to use EternalStorage in the module contracts. Even if storage layout change of the module will be necessary, all storage variables are located in 1-2 files, so it should be easy to track them down. Here, the regular rule: "Don't reorder storage variables, append new variables only to the end" should apply. 

The mediator, however, will continue to use eternal storage pattern.